tests/test_conservation.py

```diff
@@ -0,0 +1,377 @@
+        state['mass_content_of_cloud_liquid_water_in_atmosphere_layer'].loc[dict(mid_levels=slice(4, 8))] = 0.03
+        state['cloud_area_fraction_in_atmosphere_layer'].loc[dict(mid_levels=slice(4, 8))] = 1.
+        return state
+
+
+class TestRRTMGShortwaveConservationWithCloudsAndMCICA(AtmosphereEnergyConservation):
+
+    def get_component_instance(self):
+        return climt.RRTMGShortwave(mcica=True)
+
+    def modify_state(self, state):
+        state['mass_content_of_cloud_liquid_water_in_atmosphere_layer'].loc[dict(mid_levels=slice(4, 8))] = 0.03
+        state['cloud_area_fraction_in_atmosphere_layer'].loc[dict(mid_levels=slice(4, 8))] = 0.5
+        return state
+
+
+class TestRRTMGLongwaveConservationWithClouds(AtmosphereEnergyConservation):
+
+    def get_component_instance(self):
+        return climt.RRTMGLongwave()
+
+    def modify_state(self, state):
+        state['mass_content_of_cloud_liquid_water_in_atmosphere_layer'].loc[dict(mid_levels=slice(4, 8))] = 0.03
+        state['cloud_area_fraction_in_atmosphere_layer'].loc[dict(mid_levels=slice(4, 8))] = 1.
+        return state
+
+
+class TestSimplePhysicsDryConservation(AtmosphereEnergyConservation):
+
+    def get_component_instance(self):
+        return climt.SimplePhysics(boundary_layer=False,
+                                   use_external_surface_specific_humidity=True)
+
+    def modify_state(self, state):
+        state['eastward_wind'].values[:] = 3.
+        return state
+
+
+class TestSimplePhysicsConservation(AtmosphereEnergyConservation):
+
+    def get_component_instance(self):
+        return climt.SimplePhysics(boundary_layer=False)
+
+    def modify_state(self, state):
+        state['eastward_wind'].values[:] = 3.
+        return state
+
+
+class TestDryConvectionConservation(AtmosphereEnergyConservation):
+
+    def get_component_instance(self):
+        return climt.DryConvectiveAdjustment()
+
+    def modify_state(self, state):
+        unstable_level = 5
+        state['air_temperature'][:unstable_level] += 10
+        state['specific_humidity'][:unstable_level] = 0.05
+        return state
+
+    def get_quantity_amount(self, state):
+        return vertical_integral(
+            state,
+            heat_capacity_including_condensible(state['specific_humidity']).values *
+            state['air_temperature'].values)
+
+    def get_quantity_forcing(self, state):
+        return 0
+
+
+class TestDryConvectionCondensibleConservation(AtmosphereTracerConservation):
+
+    def get_component_instance(self):
+        self.tracer_name = 'specific_humidity'
+        return climt.DryConvectiveAdjustment()
+
+    def modify_state(self, state):
+        unstable_level = 5
+        state['air_temperature'][:unstable_level] += 10
+        state['specific_humidity'][:unstable_level] = 0.05
+        return state
+
+    def get_quantity_amount(self, state):
+        return vertical_integral(
+            state, state['specific_humidity'])
```

Comment:
I would say convert units here, but you don't need to since you're comparing to a forcing of zero (and there are no zero-offset units for humidity).